warsawlo/schema.py

```diff
@@ -116,6 +117,54 @@ class Meta:
         interfaces = (relay.Node,)
 
 
+class CharInFilter(django_filters.BaseInFilter, django_filters.CharFilter):
+    pass
+
+
+class SearchFilter(django_filters.FilterSet):
+    QUERY_FIELD = "school_name"
+    SCHOOL_TYPE_GENERALISED = "szkoła ponadpodstawowa"
+
+    class Meta:
+        model = School
+        fields = ["school_type", "is_public"]
+
+    query = django_filters.CharFilter(method="query_filter")
+
+    district = django_filters.CharFilter(field_name="address__district")
+
+    # TODO: Filter input as list not string
+    # or filtering
+    available_extended_subjects = CharInFilter(
+        field_name="highschoolclass__extendedsubject__name"
+    )
+
+    def query_filter(self, queryset, value):
+        return (
+            queryset.annotate(similarity=TrigramSimilarity(self.QUERY_FIELD, value))
+            .filter(similarity__gte=0.05)
```

Comment:
In the future, I suggest checking also for words that contain the searched substring, cause if the school_name is a very long string (for example full name of Batory high school) and we're searching only for "bat", the value of trigram similarity is going to be small while we still expect to get the Batory high school. But this is for another PR.

Reply:
Yeah, for now I'm leaving it as it is.